valkyrie/lib/valkyrie/value_mapper.rb

```diff
@@ -1,10 +1,31 @@
 # frozen_string_literal: true
 module Valkyrie
+  ##
+  # ValueMapper is a way to handle coordinating extendable casting of values
+  # depending on what the value is. It's used in many of the adapters in
+  # Valkyrie.
+  # @example Defining a ValueMapper which converts symbols to strings.
+  #   class ParentMapper < ValueMapper
+  #   end
+  #   class Stringify < ValueMapper
+  #     ParentMapper.register(self)
+  #     def self.handles?(value)
+  #       value.kind_of?(Symbol)
+  #     end
+  #     def result
+  #       value.to_s
+  #     end
+  #   end
+  # @example Use a ValueMapper
+  #   ParentMapper.for(:symbol).result # => "symbol"
```

Comment:
They can totally register at the ValueMapper level if they want, I think that's fine.